File: src/main/java/io/pivotal/pal/tracker/WelcomeController.java
package io.pivotal.pal.tracker;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class WelcomeController {



    private String message;


     public WelcomeController( @Value("${WELCOME_MESSAGE:HelloPim}")String message){
            this.message= message;
    }

    @GetMapping("/")
    public String sayHello() {
        return this.message;
    }
}

File: src/main/kotlin/io/pivotal/pal/tracker/WelcomeController.kt
package io.pivotal.pal.tracker

import org.springframework.beans.factory.annotation.Value
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
class WelcomeController constructor(@Value("\${WELCOME_MESSAGE:Hello World!}") message: String) {

    val message = message;

    @GetMapping("/")
    fun sayHello():String {
        return message
    }
}


